server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/pushsubscription/PushListener.scala

```diff
@@ -58,15 +59,18 @@ object PushListener {
 }
 
 class PushListener @Inject()(pushRepository: PushSubscriptionRepository,
-                   webPushClient: WebPushClient,
-                   pushSerializer: PushSerializer) extends ReactiveGroupEventListener {
+                             webPushClient: WebPushClient,
+                             pushSerializer: PushSerializer,
+                             delegationStore: DelegationStore) extends ReactiveGroupEventListener {
 
   override def getDefaultGroup: Group = PushListenerGroup()
 
   override def reactiveEvent(event: Event): Publisher[Void] =
     event match {
       case event: StateChangeEvent =>
-        SFlux(pushRepository.list(event.username))
+        SMono.just(event.username)
+          .concatWith(delegationStore.authorizedUsers(event.username))
```

Comment:
why we don't use 
`SFlux(delegationStore.authorizedUsers(event.username)`

Reply:
your solution does include the owner of the event himself

Reply:
ah, i got it